tutorials/nemo-retriever-synthetic-data-generation/main.py

```diff
@@ -0,0 +1,204 @@
+    if cfg.easiness_filter:
+        filters.append(
+            ScoreFilter(
+                EasinessFilter(
+                    cfg.base_url,
+                    cfg.api_key,
+                    cfg.easiness_filter,
+                    cfg.percentile,
+                    cfg.truncate,
+                    cfg.batch_size,
+                ),
+                text_field=["text", "question"],
+                score_field="easiness_scores",
+            )
+        )
+    if cfg.answerability_filter:
+        filters.append(
+            ScoreFilter(
+                AnswerabilityFilter(
+                    cfg.base_url,
+                    cfg.api_key,
+                    cfg.answerability_filter,
+                    cfg.answerability_system_prompt,
+                    cfg.answerability_user_prompt_template,
+                    cfg.num_criteria,
+                ),
+                text_field=["text", "question"],
+                score_field="answerability_scores",
+            )
+        )
+
+    if filters:
+        filtering_pipeline = Sequential(filters)
+    else:
+        filtering_pipeline = None
+
+    return sdg_pipeline, filtering_pipeline
+
+
+def write_to_beir(args: Any, dataset: DocumentDataset, filtered: bool = False):
+
+    df = dataset.df
+    df = df.compute()
+    if filtered:
+        save_dir = os.path.join(args.output_dir, "beir", "filtered")
+        qrels_save_dir = os.path.join(args.output_dir, "beir", "filtered", "qrels")
+        corpus_save_path = os.path.join(
+            args.output_dir, "beir", "filtered", "corpus.jsonl"
+        )
+        queries_save_path = os.path.join(
+            args.output_dir, "beir", "filtered", "queries.jsonl"
+        )
+    else:
+        save_dir = os.path.join(args.output_dir, "beir", "all")
+        qrels_save_dir = os.path.join(args.output_dir, "beir", "all", "qrels")
+        corpus_save_path = os.path.join(args.output_dir, "beir", "all", "corpus.jsonl")
+        queries_save_path = os.path.join(
+            args.output_dir, "beir", "all", "queries.jsonl"
+        )
+
+    os.makedirs(save_dir)
+    os.makedirs(qrels_save_dir)
+
+    df[["question-id", "question"]].rename(
+        columns={"question-id": "_id", "question": "text"}
+    ).to_json(queries_save_path, lines=True, orient="records")
+
+    if filtered:
+        corpus_file_path = os.path.join(args.output_dir, "beir", "all", "corpus.jsonl")
+        if os.path.exists(corpus_file_path):
+            shutil.copy(corpus_file_path, corpus_save_path)
+        else:
+            raise ValueError("Generate data first")
+    else:
+        df[["_id", "text"]].to_json(corpus_save_path, lines=True, orient="records")
+
+    df[["question-id", "_id", "score"]].rename(
+        columns={"question-id": "query-id", "_id": "corpus-id"}
+    ).to_csv(os.path.join(qrels_save_dir, "test.tsv"), sep="\t", index=False)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--input_file",
```

Comment:
There is inconsistency between using `_` and `-` to separate words. Can you unify and use `-` to follow the convention as the rest of curator?

Reply:
fixed